pook/activate_async.py

```diff
@@ -14,12 +15,14 @@ def activate_async(fn, _engine):
         function: decorator wrapper function.
     """
     @functools.wraps(fn)
-    async def wrapper(*args, **kw): # noqa
+    @coroutine
+    def wrapper(*args, **kw):
         _engine.activate()
         try:
-            coro = fn(*args, **kw)
-            if isawaitable(coro):
-                await coro
+            if iscoroutinefunction(fn):
+                yield from fn(*args, **kw)  # noqa
```

Comment:
Not sure why `# noqa` is required here.

Reply:
`make test` with python 2 will fail with 999 flake syntax error

Reply:
Dammit, I tend to about Python 2.x compability from time to time haha